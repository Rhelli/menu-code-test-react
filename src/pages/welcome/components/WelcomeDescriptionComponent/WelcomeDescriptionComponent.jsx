import React, { useState } from 'react';
import styles from './WelcomeDescriptionComponent.module.scss';

const WelcomeDescriptionComponent = () => {
  const [description, setDescription] = useState(false);

  const handleDescriptionClick = () => {
    if (description === false) {
      setDescription(true);
    }
    if (description === true) {
      setDescription(false);
    }
    return description;
  };

  return (
    <div className={description ? styles.descriptionFull : styles.description}>
      <p>The Chez Paree is the quintessential French dining experience, in the center of London&apos;s busy Soho.
      With a well earned Michelin star behind Jean-Luc Picard&apos;s Parisian gem, the <i>Chez Paree</i> is been
      critically acclaimed by numerous high-flyer critiques and restaurant-goers alike.</p>
      <button onClick={handleDescriptionClick}>+ Read More</button>
    </div>
  );
};

export default WelcomeDescriptionComponent;
