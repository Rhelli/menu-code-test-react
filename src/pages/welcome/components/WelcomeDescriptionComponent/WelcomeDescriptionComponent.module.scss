.description,
.descriptionFull {
  width: 100%;

  p {
    margin-top: 20px;
    width: 100%;
    line-height: 20px;
    color: #303030;
    overflow: hidden;
    white-space: nowrap;
    text-overflow: ellipsis;
  }

  button {
    outline: none;
    border: none;
    font-size: 12px;
    color: #da3743;
    background-color: transparent;

    &:hover {
      text-decoration: underline;
      cursor: pointer;
    }
  }
}

.description {
  p {
    height: 30px;
  }
}

.descriptionFull {
  p {
    overflow: unset;
    white-space: normal;
    text-overflow: unset;
  }
}
