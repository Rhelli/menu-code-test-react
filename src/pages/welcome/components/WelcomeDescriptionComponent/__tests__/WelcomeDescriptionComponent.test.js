import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import Enzyme, { mount } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import WelcomeDescriptionComponent from '../WelcomeDescriptionComponent';

Enzyme.configure({ adapter: new Adapter() });

const welcomeDescriptionComponent = (
  <WelcomeDescriptionComponent />
);

describe('The appearance and functionality of the WelcomeDescriptionComponent', () => {
  const component = mount(<WelcomeDescriptionComponent />);
  it('Matches the previous snapshot of the WelcomeDescriptionComponent', () => {
    const tree = renderer.create(welcomeDescriptionComponent).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Has the correct master container with the correct default class', () => {
    expect(component.childAt(0).hasClass('description')).toBe(true);
    expect(component.childAt(0).type()).toBe('div');
  });

  it('Contains the read more button', () => {
    expect(component.childAt(0).childAt(1).name()).toBe('button');
  });

  it('Displays the description content', () => {
    expect(component.childAt(0).childAt(0).name()).toBe('p');
  });
});
