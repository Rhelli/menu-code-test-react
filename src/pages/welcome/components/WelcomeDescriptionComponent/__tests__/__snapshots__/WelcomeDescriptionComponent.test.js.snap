// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The appearance and functionality of the WelcomeDescriptionComponent Matches the previous snapshot of the WelcomeDescriptionComponent 1`] = `
<div
  className="description"
>
  <p>
    The Chez Paree is the quintessential French dining experience, in the center of London's busy Soho. With a well earned Michelin star behind Jean-Luc Picard's Parisian gem, the 
    <i>
      Chez Paree
    </i>
     is been critically acclaimed by numerous high-flyer critiques and restaurant-goers alike.
  </p>
  <button
    onClick={[Function]}
  >
    + Read More
  </button>
</div>
`;
